sources/poly.cc

```diff
@@ -2561,11 +2561,23 @@ const poly poly::argument_to_poly (PHEAD WORD *e, bool with_arghead, bool sort_u
 
 	res[0] = ri;
 	
-	// normalize, since the Form order is probably not the polynomial order
-	// for multiple variables
+	// JD: For univariate cases, check whether the ordering is correct or not.
+	// There are various ways to arrive here, but with an incorrect ordering, such
+	// as interactions with collect, moving a polyratfun out of another function
+	// argument, etc.
+	// If the ordering is not correct, make sure we normalize. In multivariate cases,
+	// always normalize as before.
+	if (sort_univar == false && AN.poly_num_vars == 1) {
```

Comment:
It's a matter of taste, but `!sort_univar` is more natural than `sort_univar == false`?

Or, these lines may be written as
```c
    if (sort_univar || AN.poly_num_vars > 1 ||
        AN.poly_num_vars == 1 && res.number_of_terms() >= 2 &&
            res[2] < res.last_monomial()[2]) {
        res.normalize();
    }
```
I'm not sure whether `AN.poly_num_vars == 1` is required or not. `AN.poly_num_vars` can be 0?